rocks-lib/src/package/version.rs

```diff
@@ -234,7 +237,16 @@ impl PackageVersionReq {
             (PackageVersionReq::SemVer(version_req), PackageVersion::SemVer(semver)) => {
                 version_req.matches(&semver.version)
             }
-            (PackageVersionReq::SemVer(..), PackageVersion::DevVer(..)) => true,
+            (PackageVersionReq::SemVer(version_req), PackageVersion::DevVer(..)) => {
+                let large_version = Version {
+                    major: u64::MAX,
+                    minor: u64::MAX,
+                    patch: u64::MAX,
+                    pre: Prerelease::EMPTY,
+                    build: BuildMetadata::EMPTY,
+                };
+                version_req.matches(&large_version)
+            }
```

Comment:
suggestion: it's quite hard figuring out what this exactly does, maybe a comment could be useful?